src/python/gudhi/representations/kernel_methods.py

```diff
@@ -9,13 +9,73 @@
+    :param weight: weight function for the persistence diagram points. This function must be defined on 2D points, ie lists or numpy arrays of the form [p_x,p_y].
+    :param kernel_approx: kernel approximation class used to speed up computation. Common kernel approximations classes can be found in the scikit-learn library (such as RBFSampler for instance).
+    :returns: the persistence weighted Gaussian kernel value between persistence diagrams.
+    :rtype: float 
+    """
+    ws1 = np.array([weight(D1[j,:]) for j in range(len(D1))])
+    ws2 = np.array([weight(D2[j,:]) for j in range(len(D2))])
+    if kernel_approx is not None:
+        approx1 = np.sum(np.multiply(ws1[:,np.newaxis], kernel_approx.transform(D1)), axis=0)
+        approx2 = np.sum(np.multiply(ws2[:,np.newaxis], kernel_approx.transform(D2)), axis=0)
+        return (1./(np.sqrt(2*np.pi)*bandwidth)) * np.matmul(approx1, approx2.T)
+    else:
+        W = np.matmul(ws1[:,np.newaxis], ws2[np.newaxis,:])
+        E = (1./(np.sqrt(2*np.pi)*bandwidth)) * np.exp(-np.square(pairwise_distances(D1,D2))/(2*bandwidth*bandwidth))
+        return np.sum(np.multiply(W, E))
+
+def persistence_scale_space_kernel(D1, D2, kernel_approx=None, bandwidth=1.):
+    """
+    This is a function for computing the persistence scale space kernel value from two persistence diagrams. The persistence scale space kernel is computed by adding the symmetric to the diagonal of each point in each persistence diagram, with negative weight, and then convolving the points with a Gaussian kernel. See https://www.cv-foundation.org/openaccess/content_cvpr_2015/papers/Reininghaus_A_Stable_Multi-Scale_2015_CVPR_paper.pdf for more details.
+    :param D1: (n x 2) numpy.array encoding the (finite points of the) first diagram. Must not contain essential points (i.e. with infinite coordinate).
+    :param D2: (m x 2) numpy.array encoding the second diagram.
+    :param bandwidth: bandwidth of the Gaussian kernel with which persistence diagrams will be convolved
+    :param kernel_approx: kernel approximation class used to speed up computation. Common kernel approximations classes can be found in the scikit-learn library (such as RBFSampler for instance).
+    :returns: the persistence scale space kernel value between persistence diagrams.
+    :rtype: float 
+    """
+    DD1 = np.concatenate([D1, D1[:,[1,0]]], axis=0)
+    DD2 = np.concatenate([D2, D2[:,[1,0]]], axis=0)
+    weight_pss = lambda x: 1 if x[1] >= x[0] else -1
+    return 0.5 * persistence_weighted_gaussian_kernel(DD1, DD2, weight=weight_pss, kernel_approx=kernel_approx, bandwidth=bandwidth)
+
+def pairwise_persistence_diagram_kernels(X, Y=None, metric="sliced_wasserstein", **kwargs):
+    """
+    This function computes the kernel matrix between two lists of persistence diagrams given as numpy arrays of shape (nx2).
+    :param X: first list of persistence diagrams. 
```

Comment:
The doc looks horrible this way, it doesn't interpret this line as describing a parameter. Maybe skipping a line before would help.
If you have trouble building the doc on your computer, it is visible online: at the bottom of the pull request: "show all tests", "Details" next to the test called "build", "python", "artifacts" (if you don't see any link called artifacts, go to circleci.com, log in using github, and start again).
I notice you changed from Google-style to sphinx-style. Both work, I am just surprised by the difference.

Reply:
Yes, you are right. Even I don't know exactly why I changed the doc style ;-) I made everything consistent.